tee_supplicant/src/tee_supplicant.c

```diff
@@ -248,7 +252,10 @@ static int shm_alloc(const struct device *dev, uint32_t num_params,
 		return TEEC_ERROR_BAD_PARAMETERS;
 	}
 
-	/*TODO: check for TEE_GEN_CAP_REG_MEM */
+	if (!(arg->gen_caps & TEE_GEN_CAP_REG_MEM)) {
```

Comment:
Done

Reply:
As you have added check in init stage and it will fail if gen_caps doesn't have CAP_REG_MEM - this code is unlikely to be reached so I don't see any point of this check here. Also, there should be easier to add static shmem support in future if there is no check here. 
So, I think this check should be removed

Reply:
Done